File: 0038-count-and-say/0038-count-and-say.ts
function countAndSay(n: number): string {
  if (n === 1) return '1'
  
  function convert(s: string): string {
    let [ans, currCh, currLen] = ['', '', 0]
    for (let i = 0; i < s.length; i++) {
      if (s[i] === currCh) {
        currLen++
      } else {
        if (currLen) ans += currLen.toString() + currCh
        currCh = s[i]
        currLen = 1
      }
    }
    if (currLen) ans += currLen.toString() + currCh
    return ans
  }
  
  return convert(countAndSay(n - 1))
}

File: 0038-count-and-say/0038-count-and-say.go
func countAndSay(n int) string {
  if n == 1 {
    return "1"
  }
  return convert(countAndSay(n-1))
}

func convert(s string) string {
  ans, currCh, currLen := "", "", 0
  for i:=0;i<len(s);i++ {
    sub := s[i:i+1]
    if sub == currCh {
      currLen++
    } else {
      if currLen > 0 {
        ans += strconv.Itoa(currLen)
        ans += currCh
      }
      currCh = sub
      currLen = 1
    }
  }
  if currLen > 0 {
    ans += strconv.Itoa(currLen)
    ans += currCh
  }
  return ans
}